uk.co.compucorp.civicrm.hrcore/tests/phpunit/CRM/HRCore/Service/CiviHRStatsGathererTest.php

```diff
@@ -1,35 +1,296 @@
+    $this->fabricateLeaveRequest($contactID, '-260 days', '-256 days');
+
+    $stats = $this->getGatherer()->gather();
+
+    $this->assertEquals(9, $stats->getEntityCount('leaveRequest'));
+    $this->assertEquals(3, $stats->getEntityCount('leaveRequestInLast100Days'));
+  }
+
+  public function testContactSubtypeCountWillMatchExpectedCount() {
+    $existingContactCount = $this->getCurrentContactCount();
+    $existingOrganizationCount = $this->getCurrentContactCount(
+      ['contact_type' => 'Organization']
+    );
+    $existingIndividialCount = $this->getCurrentContactCount(
+      ['contact_type' => 'Individual']
+    );
+
+    ContactTypeFabricator::fabricate(['name' => 'Cat']);
+    ContactTypeFabricator::fabricate(['name' => 'Dog']);
+
+    ContactTypeFabricator::fabricate([
+      'name' => 'CatClub',
+      'parent_id' => 'Organization'
+    ]);
+
+    ContactTypeFabricator::fabricate([
+      'name' => 'DogClub',
+      'parent_id' => 'Organization'
+    ]);
+
+    ContactFabricator::fabricate([
+      'contact_sub_type' => 'Cat',
+      'first_name' => 'Mittens',
+      'last_name' => 'Meow'
+    ]);
+
+    ContactFabricator::fabricate([
+      'contact_sub_type' => 'Cat',
+      'first_name' => 'Felix',
+      'last_name' => 'da Housecat'
+    ]);
+
+    ContactFabricator::fabricate([
+      'contact_sub_type' => 'CatClub',
+      'contact_type' => 'Organization',
+      'organization_name' => 'Kool Katz Klub',
+    ]);
+
+    ContactFabricator::fabricate([
+      'contact_sub_type' => 'Dog',
+      'first_name' => 'Lassie',
+      'last_name' => 'Come Home'
+    ]);
+
+    ContactFabricator::fabricate([
+      'contact_sub_type' => 'DogClub',
+      'contact_type' => 'Organization',
+      'organization_name' => 'Dope Dogz',
+    ]);
+
+    $stats = $this->getGatherer()->gather();
+
+    $expectedContacts = $existingContactCount + 5;
+    $expectedOrgs = $existingOrganizationCount + 2;
+    $expectedIndividuals = $existingIndividialCount + 3;
+    $organizationCount = $stats->getContactSubtypeCount('Organization');
+    $individualCount = $stats->getContactSubtypeCount('Individual');
+
+    $this->assertEquals(1, $stats->getContactSubtypeCount('DogClub'));
+    $this->assertEquals(1, $stats->getContactSubtypeCount('Dog'));
+    $this->assertEquals(1, $stats->getContactSubtypeCount('CatClub'));
+    $this->assertEquals(2, $stats->getContactSubtypeCount('Cat'));
+    $this->assertEquals($expectedOrgs, $organizationCount);
+    $this->assertEquals($expectedIndividuals, $individualCount);
+    $this->assertEquals($expectedContacts, $stats->getEntityCount('contact'));
+  }
+
+  public function testDeletedEntitiesWillNotBeIncluded() {
+    $existingContactCount = $this->getCurrentContactCount();
+    $deletedContact = ContactFabricator::fabricate();
+    $deletedContactID = $deletedContact['id'];
+    civicrm_api3('Contact', 'delete', ['id' => $deletedContactID]);
```

Comment:
Shouldn't we check the same thing for all the other entities which also have the `is_deleted` field?

Reply:
I didn't because I figured then we're not testing specifics of the gatherer class, but instead testing that each other entity API will behave correctly (and partly because I was lazy). I can add tests for all entities.